python/tests/compat/conftest.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+import subprocess
+import os
+import venv
+import tempfile
+
+class ErrorInVenv(Exception):
+    """To signal errors occuring from within the venv"""
+
+def run_shell_commands(commands_list, cwd=None):
+    command = " && ".join(commands_list)
+    print("Executing command: ", command)
+    result = subprocess.run(command, cwd=cwd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Do we need `shell=True`? If not, `check_call` is more idiomatic

Reply:
Reworked to make it work on windows as well. Now `shell=True` is required for windows.

Also [python docs](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module) say:
```
The recommended approach to invoking subprocesses is to use the [run()](https://docs.python.org/3/library/subprocess.html#subprocess.run) function for all use cases it can handle. For more advanced use cases, the underlying [Popen](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) interface can be used directly.
```

Reply:
Aha, my knowledge is out of date